packages/angular/src/generators/ng-add/utilities/workspace.ts

```diff
@@ -49,35 +51,95 @@ export function createNxJson(
   const { newProjectRoot = '' } = readJson(tree, 'angular.json');
   const { npmScope } = options;
 
+  const targets = getWorkspaceCommonTargets(tree);
+
   writeJson<NxJsonConfiguration>(tree, 'nx.json', {
     ...(npmScope ? { npmScope } : {}),
     affected: {
       defaultBase: options.defaultBase ?? deduceDefaultBase(),
     },
-    implicitDependencies: {
-      'package.json': {
-        dependencies: '*',
-        devDependencies: '*',
-      },
-      '.eslintrc.json': '*',
-    },
     tasksRunnerOptions: {
       default: {
         runner: 'nx/tasks-runners/default',
         options: {
-          cacheableOperations: ['build', 'lint', 'test', 'e2e'],
+          cacheableOperations: [
+            'build',
+            targets.test ? 'test' : undefined,
+            targets.lint ? 'lint' : undefined,
+            targets.e2e ? 'e2e' : undefined,
+          ].filter(Boolean),
         },
       },
     },
+    namedInputs: {
+      sharedGlobals: [],
+      default: ['{projectRoot}/**/*', 'sharedGlobals'],
+      production: [
+        'default',
+        ...(targets.test
+          ? [
+              '!{projectRoot}/tsconfig.spec.json',
+              '!{projectRoot}/**/*.spec.[jt]s',
+              '!{projectRoot}/karma.conf.js',
+            ]
+          : []),
+        targets.lint ? '!{projectRoot}/.eslintrc.json' : undefined,
```

Comment:
Shouldn't we only add `.eslintrc.json` here if it exists?

Reply:
Same as previous.